scripts/sysman/firmware.yml

```diff
@@ -90,6 +90,49 @@ returns:
     - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
         - "User does not have permissions to perform this operation."
 --- #--------------------------------------------------------------------------
+type: function
```

Comment:
@vaidyan1-int I have a query regarding this(Not sure if its worth to discuss, but just want to highlight so that we dont miss any corner case).
Suppose user first called zesFirmwareFlash, and then user called this UpdateProgress.

But as zesFirmwareFlash is not aware that UpdateProgress might be called. So the moment zesFirmwareFlash is called it will immediately flash the firmware. Now if user called UpdateProgress , then i think progress percentage would in max cases start with percentage as greater than 0%.
So,  is it okay?

Reply:
Also could you please tell what is the use case of this API UpdateProgress?
Would user call this UpdateProgress in a loop to get continous update?

Isnt it possible to enhance zesFirmwareFlash API by adding one more parameter named pCompletionPercent?

Reply:
@jitendrasharma1989 I'd prefer not to combine existing FirmwareFlash API with completion percentage as one is write operation and other is read. Better to decouple

Reply:
@aravindksg, I am still trying to understand how user would use this updateProgress API?
Is it like from one thread user would invoke a blocking call updateProgress ()
And in another thread user would invoke zesFirmwareFlash(). If that is the case then updateProgress could tell about firmware update progress nicely. But if that is the case then may be we need to add some timeout parameter in updateProgress  API. This timeout parameter would help in letting Sysman know that, some firmware flash command is about to come in this timeout.

Reply:
so FirmwareFlash is a non-blocking call, the intention of this API is to allow user to immediately call back in same thread for progress update after calling zesFirmwareFlash. So user can call this in a loop at app level. We could add pseudo code in prog guide for this.

Reply:
@aravindksg   the earlier GPU platforms , FirmwareFlash is blocking call and app need to wait till its done. Is it going to change ?

Reply:
@saik-intel not seeing anywhere in spec where we have mentioned FirmwareFlash is blocking call. Generally though the APIs are non-blocking unless we have some timeout param

Reply:
if FirmwareFlash is not blocking, then we should add that to the documentation. I would have expected it to be blocking, and if not, then documentation to mention how user can confirm completion (which might be using getFlashProgress.

Reply:
Updated description to include this